util/tls/tls_socket.cc

```diff
@@ -165,6 +165,24 @@ auto TlsSocket::Close() -> error_code {
   return next_sock_->Close();
 }
 
+class SpinCounter {
+ public:
+  explicit SpinCounter(size_t limit) : limit_(limit) {
+  }
+
+  void Check(bool condition) {
+    current_it_ = condition ? current_it_ + 1 : 0;
+    if (current_it_ >= limit_) {
+      LOG(WARNING) << "IO loop spin limit reached: " << limit_;
```

Comment:
this warning will appear at line 176 instead of appearing in the loop of the function that is deadlocked.

Reply:
GOOOD catch! Will address this